src/components/utils/popover.ts

```diff
@@ -381,37 +443,229 @@ export default class Popover extends EventsDispatcher<PopoverEvent> {
+    }
+    clickedItem.onActivate(clickedItem, event);
+
+    if (clickedItem.closeOnActivate) {
+      this.hide();
+    } else {
+      /** If popover is not intended to close once item is activated, we need to update active state of activated item */
+      clickedItem.isActive = !clickedItem.isActive;
+      itemEl.classList.toggle(Popover.CSS.itemActive);
+    }
   }
 
   /**
-   * Creates Flipper instance to be able to leaf tools
+   * Enables confirmation state for specified item.
+   * Replaces item element in popover so that is becomes highlighted in a special way
+   *
+   * @param item - item to enable confirmation state for
+   * @param itemEl - html element corresponding to the item
+   * @param itemIndex - index of the item in all items list
    */
-  private enableFlipper(): void {
-    const tools = Array.from(this.nodes.wrapper.querySelectorAll(`.${Popover.CSS.item}`)) as HTMLElement[];
-
-    this.flipper = new Flipper({
-      items: tools,
-      focusedItemClass: Popover.CSS.itemFocused,
-      allowedKeys: [
-        keyCodes.TAB,
-        keyCodes.UP,
-        keyCodes.DOWN,
-        keyCodes.ENTER,
-      ],
-    });
+  private enableConfirmationStateForItem(item: PopoverItemWithConfirmation, itemEl: HTMLElement, itemIndex: number): void {
+    /** Save root item requiring confirmation to restore original state on popover hide */
+    if (this.itemsRequiringConfirmation[itemIndex] === undefined) {
+      this.itemsRequiringConfirmation[itemIndex] = item;
+    }
+    const newItemData = {
+      ...item,
+      ...item.confirmation,
+      confirmation: item.confirmation.confirmation,
+    } as PopoverItem;
+
+    this.items[itemIndex] = newItemData;
+
+    const confirmationStateItemEl = this.createItem(newItemData as PopoverItem);
+
+    confirmationStateItemEl.classList.add(Popover.CSS.itemConfirmation, ...Array.from(itemEl.classList));
+    itemEl.parentElement.replaceChild(confirmationStateItemEl, itemEl);
+
+    this.enableSpecialHoverAndFocusBehavior(confirmationStateItemEl);
+
+    this.reactivateFlipper(
+      this.flippableElements,
+      this.flippableElements.indexOf(confirmationStateItemEl)
+    );
+  }
+
+  /**
+   * Brings specified element corresponding to popover item to its original state
+   *
+   * @param itemEl - item in confirmation state
+   */
+  private cleanUpConfirmationStateForItem(itemEl: Element): void {
+    const allItems = Array.from(this.nodes.items.children);
+    const index = allItems.indexOf(itemEl);
+
+    const originalItem = this.itemsRequiringConfirmation[index];
+
+    if (originalItem === undefined) {
+      return;
+    }
+    const originalStateItemEl = this.createItem(originalItem);
+
+    itemEl.parentElement.replaceChild(originalStateItemEl, itemEl);
+    this.items[index] = originalItem;
+
+    delete this.itemsRequiringConfirmation[index];
+
+    itemEl.removeEventListener('mouseleave', this.removeSpecialHoverBehavior);
+    this.flipper.removeOnFlip(this.onFlip);
```

Comment:
it should be moved to `disableSpecialHoverAndFocusBehavior` to make it opposite the `enableSpecialHoverAndFocusBehavior`

Reply:
added `disableSpecialHoverAndFocusBehavior`